server/src/serializers/MovieSerializer.js

```diff
@@ -1,16 +1,22 @@
 import ReviewSerializer from "./ReviewSerializer.js";
 
 class MovieSerializer {
-  static getDetails(movies) {
+  static async getDetails(movies) {
     const allowedAttributes = ["id", "title"];
 
-    const serializedMovies = movies.map((movie) => {
-      let serializedMovie = {};
-      for (const attribute of allowedAttributes) {
-        serializedMovie[attribute] = movie[attribute];
-      }
-      return serializedMovie;
-    });
+    const serializedMovies = await Promise.all(
+      movies.map(async (movie) => {
+        let serializedMovie = {};
+        for (const attribute of allowedAttributes) {
+          serializedMovie[attribute] = movie[attribute];
+        }
+
+        let averageRating = await movie.$relatedQuery("reviews").avg("rating");
+        serializedMovie.averageRating = parseFloat(averageRating[0].avg).toFixed(1);
```

Comment:
make this a method on the Movie class so that we can get the `.averageRating` for any movie, not just in the serializer!